extensions/amp-recaptcha-input/0.1/amp-recaptcha-service.js

```diff
@@ -151,30 +165,30 @@ export class AmpRecaptchaService {
 
   /**
    * Function to create our recaptcha boostrap iframe.
-   * @param {!Element} element
+   * @param {string} sitekey
    * @private
    */
-  initialize_(element) {
-
-    /* the third parameter 'recaptcha' ties it to the 3p/recaptcha.js */
-    this.iframe_ = getIframe(this.win_, element, 'recaptcha');
-
-    this.unlisteners_ = [
-      this.listenIframe_(
-          MESSAGE_TAG + 'ready', this.recaptchaApiReady_.resolve
-      ),
-      this.listenIframe_(
-          MESSAGE_TAG + 'token', this.tokenMessageHandler_.bind(this)
-      ),
-      this.listenIframe_(
-          MESSAGE_TAG + 'error', this.errorMessageHandler_.bind(this)
-      ),
-    ];
-    this.executeMap_ = {};
+  initialize_(sitekey) {
+    return this.createRecaptchaFrame_(sitekey).then(iframe => {
+      this.iframe_ = iframe;
+
+      this.unlisteners_ = [
+        this.listenIframe_(
+            'amp-recaptcha-ready', () => this.recaptchaApiReady_.resolve()
+        ),
+        this.listenIframe_(
+            'amp-recaptcha-token', this.tokenMessageHandler_.bind(this)
+        ),
+        this.listenIframe_(
+            'amp-recaptcha-error', this.errorMessageHandler_.bind(this)
+        ),
+      ];
+      this.executeMap_ = {};
 
-    this.iframe_.classList.add('i-amphtml-recaptcha-iframe');
-    this.body_.appendChild(this.iframe_);
-    this.iframeLoadPromise_ = loadPromise(this.iframe_);
+      this.iframe_.classList.add('i-amphtml-recaptcha-iframe');
```

Comment:
is it possible to move this line to `createRecaptchaFrame_`? it makes more sense to me to group the setAttribute, set class name logic together.